script.js

```diff
@@ -0,0 +1,690 @@
+
+const tradeCards = () => {
+  // trade the selected cards in playerhand
+  for (let i = 0; i < playerHand.length; i += 1) {
+    for (let j = 0; j < selectedCardsArray.length; j += 1) {
+      if (selectedCardsArray[j].rank === playerHand[i].rank
+        && selectedCardsArray[j].suit === playerHand[i].suit) {
+        playerHand.splice(i, 1, deck.pop());
+      }
+    }
+  }
+  console.log(playerHand);
+  // empty the cards in trade array after the trade is done
+  selectedCardsArray = [];
+  // clear display of current player's hand
+  cardContainer.innerHTML = '';
+  for (let i = 0; i < playerHand.length; i += 1) {
+    const cardEl = createCard(playerHand[i]);
+    cardContainer.appendChild(cardEl);
+  }
+};
+
+// appending child order/display order
+mainDisplayDiv.appendChild(bidAmountLabel);
+mainDisplayDiv.appendChild(bidAmountInput);
+mainDisplayDiv.appendChild(brEL);
+mainDisplayDiv.appendChild(brEL.cloneNode());
+mainDisplayDiv.appendChild(dealButton);
+mainDisplayDiv.appendChild(brEL.cloneNode());
+mainDisplayDiv.appendChild(cardContainer);
+// store the cards based on rank, each rank has a different row in the array
+let rankHandArray = [[]];
+
+const arrangeCards = () => {
+  // for each position of the playerhand starting from the 0th index
+  // find higher rank cards and swap them in those position
+  let j = 1;
+  for (let i = 0; i < (playerHand.length - 1); i += 1) {
+    for (let k = j; k < playerHand.length; k += 1) {
+      if (playerHand[k].rank > playerHand[i].rank) {
+        const lowerRankCard = playerHand[i];
+        const higherRankCard = playerHand[k];
+        playerHand[i] = higherRankCard;
+        playerHand[k] = lowerRankCard;
+      }
+    }
+    j += 1;
+  }
+};
+const groupRankedPlayerCards = () => {
+  // empty hand from previous games
+  // playerHand[0].rank = 13;
+  // playerHand[1].rank = 12;
+  // playerHand[2].rank = 11;
+  // playerHand[3].rank = 10;
+  // playerHand[4].rank = 1;
+  // playerHand[0].suit = 'diamonds';
+  // playerHand[1].suit = 'diamonds';
+  // playerHand[2].suit = 'diamonds';
+  // playerHand[3].suit = 'diamonds';
+  // playerHand[4].suit = 'diamonds';
+  // arrangeCards();
+  rankHandArray = [[]];
+  rankHandArray[0].push(playerHand[0]);
+  let rankRow = 0;
+  for (let i = 1; i < playerHand.length; i += 1) {
+    // store the current card in the row array containing same rank cards
+    if (playerHand[i].rank === playerHand[i - 1].rank) {
+      rankHandArray[rankRow].push(playerHand[i]);
+    } else { // store the current card in the next row array if different rank
+      rankHandArray.push([]);
+      rankRow += 1;
+      rankHandArray[rankRow].push(playerHand[i]);
+    }
+  }
+};
+let rankHandArrayLength = 0;
+let straightFlush = null;
+let fourOfAKind = null;
+let fullHouse = null;
+let flush = null;
+let straights = null;
+let straightsToAce = null;
+let threeOfAKind = null;
+let twoPairs = null;
+let pair = null;
+let highCard = null;
+let royalFlush = null;
```

Comment:
Unless you explicitly want to set a `null` value, you can just declare these variables without the assignment, e.g. `let straightFlush;` if you want to declare but don't want to initialise the var yet.

Reply:
These are also not the best variable names. It's not clear to me from the name what you plan to store inside these variables. See naming conventions here: https://swe1.rocketacademy.co/course-logistics/naming-casing-and-commenting-conventions#naming